### PR TITLE
ref :Lifting state in dashboard

### PR DESCRIPTION
**Pull request by Nassimancer**
One Line Description
Lifting State in Dashboard
Requirements
Create handler function(s) in the parent component to avoid setting state in the child component. The state should be managed in the parent component
Test Steps
Check if the navigation in the sidebar works correctly
Checklist

- [ ] Pull Request title includes the issue number and a brief description of the change.
- [ ] All changes should be tested and verified to work correctly.
- [ ] Code follows frontend best practices.
- [ ] Branch names follow the conventions in the contributing file.
- [ ] Commit messages follow the naming conventions in the contributing file.

### CODE REVIEW

File: src/pages/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -10,9 +10,13 @@ import styles from './Dashboard.module.css';
...
   const [title, setTitle] = useState('Dashboard');
 
+  const handleTitleChange = (newTitle) => {
+    setTitle(newTitle);
+  };
+
   return (
     <div className={styles.dashboard}>
-      <Sidebar setTitle={setTitle} />
+      <Sidebar setTitle={handleTitleChange} />
```
Comment (AnnaSolovykh) [inline src/pages/Dashboard/Dashboard.jsx:19]: It is better to use `onTitleChange` for the prop name when passing it to the child component. 

<Sidebar onTitleChange ={handleTitleChange} />

**Review by AnnaSolovykh — CHANGES_REQUESTED** [review]
This is small remark - please adjust Sidebar component accordingly with the new prop name.

Overall: **great job, you got it!** I was so happy to see your PR, happy you've made it!

**Comment by AnnaSolovykh** [comment]
The prop name should be called **onTitleChange**. This **prop** is what you send to the child component.

The function should be named using the **handle** prefix, such as handleSomething (e.g., **handleClick**, **handleTitleChange**). The prop should be called using the **on** prefix, such as **onClick**, **onTitleChange**.

`onTitleChange={handleTitleChange}`

In this example, **onTitleChange** is a prop, and **handleTitleChange** is a handler function.

Why is it best practice to use the **on** prefix for prop names? This is because you're **attaching a listener to the function, indicating that this function is responsible for handling an event.**

_The on prefix is a widely recognized convention in event handling. By using onTitleChange, it signals to anyone reading your code that this prop is an event handler. Using on for props and handle for functions helps maintain a clear separation between the component's interface (props) and its internal implementation (handler functions). It indicates that the prop (onTitleChange) is a callback function meant to handle events triggered by the child component._

**Review by AnnaSolovykh — APPROVED** [review]
Well done! Spongebob Many Thumbs

**Review by ElenaCherpakova — APPROVED** [review]

**Pull request merged**